pyiceberg/manifest.py

```diff
@@ -308,6 +308,7 @@ def data_file_with_partition(partition_type: StructType, format_version: Literal
             field_id=field.field_id,
             name=field.name,
             field_type=partition_field_to_data_file_partition_field(field.field_type),
+            required=False,
```

Comment:
What do you think of carrying this over from the field?

```suggestion
            required=field.required,
```

I would prefer this because Avro schema's are translated to Iceberg, and then used in reading the files. In Avro reading an optional field is different than reading a required field (in the case of an optional field, it will first read a boolean checking if the value is there).

Reply:
Sounds good @Fokko . I've made the corresponding change to partition_type to carry this over from the partition field as suggested 👍